### PR TITLE
Make the user specify the path to the flutter repo in Xcode

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -502,6 +502,7 @@
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
 				FLUTTER_APPLICATION_PATH = path/to/application;
+				FLUTTER_PATH = path/to/flutter/repo;
```
Comment (abarth) [inline sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj]: We've been calling this the FLUTTER_ROOT to distinguish it from the path to the flutter package, which is inside the flutter root.

**Comment by abarth** [comment]
LGTM

File: sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj (hunk 1 above)
Comment (chinmaygarde) [inline sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj]: Renamed to FLUTTER_ROOT

File: sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -502,6 +502,7 @@
 				DYLIB_CURRENT_VERSION = 1;
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
 				FLUTTER_APPLICATION_PATH = path/to/application;
+				FLUTTER_ROOT = path/to/flutter/repo;
```
Comment (krisgiesing) [inline sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj]: Is it not possible to derive this from the project location?

Comment (chinmaygarde) [inline sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj]: The Xcode project will presumably be moved around by the user (so stuff like launch screens, icons, entitlements etc..) can be added. Also, the SDK is generated from the flutter/engine repo entirely and does not need flutter/flutter at build time.

Comment (krisgiesing) [inline sky/build/sdk_xcode_harness/FlutterApplication.xcodeproj/project.pbxproj]: Ah, that makes sense.

**Comment by chinmaygarde** [comment]
Wiki updated https://github.com/flutter/engine/wiki/Flutter-Apps-on-iOS

**Pull request merged**